### PR TITLE
libxml2: Migrate to CMake

### PR DESCRIPTION
**Pull request by hmelder**

### CODE REVIEW

**Comment by hmelder** [comment]
#47

File: phases/16-libxml2.bat [hunk 1]
```diff
@@ -13,46 +13,31 @@ for /f "usebackq delims=" %%i in (`call %BASH% '../scripts/get-latest-github-rel
...
 echo ### Installing
-:: rename libxml2_a.lib to xml2.lib to allow linking using -lxml2
-:: (the wildcard suffix is required to suppress the "file or directory" prompt)
-xcopy /Y /F "bin.msvc\libxml2_a.lib" "%INSTALL_PREFIX%\lib\xml2.lib*" || exit /b 1
-xcopy /Y /F "%SRCROOT%\%PROJECT%\include\libxml\*.h" "%INSTALL_PREFIX%\include\libxml\" || exit /b 1
-
-:: write pkgconfig file
-call "%~dp0\..\scripts\common.bat" write_pkgconfig libxml-2.0 %TAG% -DLIBXML_STATIC -lxml2 "%LIBS_PRIVATE%" || exit /b 1
+:: libxml2-export.cmake is generated while installing libxml2
+ninja install
```
Comment (triplef) [inline phases/16-libxml2.bat:43]:
```suggestion
ninja install || exit /b 1
```

File: phases/17-libxslt.bat [hunk 2]
```diff
@@ -27,6 +27,7 @@ cscript configure.js ^
   %CONFIGURE_OPTS% ^
   "prefix=%INSTALL_PREFIX%" ^
   "include=%INSTALL_PREFIX%\include" ^
+  "include=%INSTALL_PREFIX%\include\libxml2" ^
```
Comment (triplef) [inline phases/17-libxslt.bat:30]: Why is this needed? Does the libxml2 include path change with this PR?

File: phases/16-libxml2.bat [hunk 3]
```diff
@@ -13,46 +13,31 @@ for /f "usebackq delims=" %%i in (`call %BASH% '../scripts/get-latest-github-rel
...
+echo ### Running cmake
+cmake .. %CMAKE_OPTIONS% ^
+-D BUILD_SHARED_LIBS=NO ^
+-D LIBXML2_WITH_LZMA=NO ^
+-D LIBXML2_WITH_PYTHON=NO ^
+-D LIBXML2_WITH_ZLIB=NO ^
+-D LIBXML2_WITH_TESTS=NO ^
+-D LIBXML2_WITH_PROGRAMS=NO ^
+-D LIBXML2_WITH_ICU=YES ^
+-D ICU_INCLUDE_DIR=%UM_INCLUDE_DIR% ^
```
Comment (triplef) [inline phases/16-libxml2.bat:33]:
```suggestion
  -D BUILD_SHARED_LIBS=NO ^
  -D LIBXML2_WITH_LZMA=NO ^
  -D LIBXML2_WITH_PYTHON=NO ^
  -D LIBXML2_WITH_ZLIB=NO ^
  -D LIBXML2_WITH_TESTS=NO ^
  -D LIBXML2_WITH_PROGRAMS=NO ^
  -D LIBXML2_WITH_ICU=YES ^
  -D ICU_INCLUDE_DIR=%UM_INCLUDE_DIR% ^
  || exit /b 1
```

File: phases/17-libxslt.bat (hunk 2 above)
Comment (hmelder) [reply phases/17-libxslt.bat:30]: libxml2 CMake installs libxml/*.h into PREFIX/include/libxml2 instead of just PREFIX/include. The old libxslt configure.js does not seem to use pkgconfig to retrieve the CFLAGs from libxml2.

File: patches/libxml2-windows-icu.patch [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
++++ b/win32/Makefile.msvc
+@@ -64,7 +64,7 @@ LIBS = $(LIBS) iconv.lib
+ !if "$(STATIC)" == "1"
+ LIBS = $(LIBS) advapi32.lib sicuuc.lib sicuin.lib sicudt.lib
+ !else
+-LIBS = $(LIBS) icuuc.lib icuin.lib icudt.lib
++LIBS = $(LIBS) icu.lib
+ !endif
+ !endif
+ !if "$(WITH_ZLIB)" == "1"
```
Comment (hmelder) [inline patches/libxml2-windows-icu.patch:59]: We can probably remove the old patch here

**Comment by triplef** [comment]
Thanks! I added a couple small comments.

What was the reason for requiring Windows-provided ICU with this change? I think it’s fine to do this, but then we should also remove the ICU phase script, and probably mention this requirement further up top in the readme (not in the "building the toolchain" section as it’s both a build- and runtime requirement.

**Comment by hmelder** [comment]
> What was the reason for requiring Windows-provided ICU with this change?

Just out of convenience. I can change it back, but this would require testing correct detection of our ICU build. Should I look into that?

File: phases/17-libxslt.bat (hunk 2 above)
Comment (triplef) [reply phases/17-libxslt.bat:30]: I see. I guess ideally we’d also update libxslt to CMake, but this seems fine for now. 👍

**Comment by triplef** [comment]
Ok makes sense. I’ve pushed these changes removing ICU to make this more explicit. Can you please rebase onto the latest master?
